pkg/executor/infoschema_reader.go

```diff
@@ -3453,6 +3457,58 @@ func (e *memtableRetriever) setDataFromKeywords() error {
 	return nil
 }
 
+func (e *memtableRetriever) setDataFromIndexUsage(ctx sessionctx.Context, schemas []*model.DBInfo) {
+	dom := domain.GetDomain(ctx)
+	rows := make([][]types.Datum, 0, 100)
+	checker := privilege.GetPrivilegeManager(ctx)
+
+	for _, schema := range schemas {
```

Comment:
Do we also need to consider pushdown predicate to make this InformationSchema table work well when there are a huge number tables. like we do for saas customer: https://github.com/pingcap/tidb/pull/50779